mllib/src/main/scala/org/apache/spark/ml/recommendation/ALS.scala

```diff
@@ -241,11 +261,17 @@ class ALSModel private[ml] (
         Float.NaN
       }
     }
-    dataset
+    val predictions = dataset
       .join(userFactors, dataset($(userCol)) === userFactors("id"), "left")
       .join(itemFactors, dataset($(itemCol)) === itemFactors("id"), "left")
       .select(dataset("*"),
         predict(userFactors("features"), itemFactors("features")).as($(predictionCol)))
+    $(unknownStrategy) match {
+      case ALSModel.Drop =>
```

Comment:
For my own curiosity, is there a reason we can't just match on the string value and avoid creating these new vals? Seems like I have seen both ways in the code.